File: massenger/app/src/main/java/com/example/massenger/ProfileViewModel.java
package com.example.massenger;

import android.util.Log;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ProfileViewModel extends ViewModel {
    
    private static final String TAG = "ProfileViewModel";
    
    private MutableLiveData<String> userName = new MutableLiveData<>("Всеволод Лазебный");
    private MutableLiveData<String> userStatus = new MutableLiveData<>("🟢 Онлайн");
    private MutableLiveData<String> userEmail = new MutableLiveData<>("vllazebnyi@example.com");
    
    public ProfileViewModel() {
        Log.d(TAG, "ProfileViewModel создан");
    }
    
    @Override
    protected void onCleared() {
        super.onCleared();
        Log.d(TAG, "ProfileViewModel уничтожен");
    }
    public LiveData<String> getUserName() {
        return userName;
    }
    public LiveData<String> getUserStatus() {
        return userStatus;
    }
    public LiveData<String> getUserEmail() {
        return userEmail;
    }
    public void setUserName(String name) {
        Log.d(TAG, "Установлено имя: " + name);
        userName.setValue(name);
    }
    public void setUserStatus(String status) {
        Log.d(TAG, "Установлен статус: " + status);
        userStatus.setValue(status);
    }
    public void setUserEmail(String email) {
        Log.d(TAG, "Установлен email: " + email);
        userEmail.setValue(email);
    }
}

File: K3439/Дмитриев Андрей/lab2/app/src/main/java/com/example/messenger/viewmodel/ProfileViewModel.kt
package com.example.messenger.viewmodel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

/**
 * ViewModel для экрана Профиля
 * Хранит состояние профиля пользователя и переживает повороты экрана
 */
class ProfileViewModel : ViewModel() {
    
    private val TAG = "ProfileViewModel"
    
    // LiveData для имени пользователя
    private val _userName = MutableLiveData<String>()
    val userName: LiveData<String> = _userName
    
    // LiveData для статуса пользователя
    private val _userStatus = MutableLiveData<String>()
    val userStatus: LiveData<String> = _userStatus
    
    init {
        Log.d(TAG, "init: ProfileViewModel создан")
        // Устанавливаем значения по умолчанию
        _userName.value = "Дмитриев Андрей"
        _userStatus.value = "Android разработчик. Люблю создавать крутые приложения!"
    }
    
    /**
     * Обновление имени пользователя
     */
    fun updateUserName(name: String) {
        Log.d(TAG, "updateUserName: Имя обновлено на '$name'")
        _userName.value = name
    }
    
    /**
     * Обновление статуса пользователя
     */
    fun updateUserStatus(status: String) {
        Log.d(TAG, "updateUserStatus: Статус обновлен на '$status'")
        _userStatus.value = status
    }
    
    override fun onCleared() {
        super.onCleared()
        Log.d(TAG, "onCleared: ProfileViewModel уничтожен")
    }
}
